src/compiler/commandLineParser.ts

```diff
@@ -141,6 +141,14 @@ namespace ts {
             category: Diagnostics.Advanced_Options,
             description: Diagnostics.Show_verbose_diagnostic_information
         },
+        {
+            name: "generateCPUProfile",
+            type: "string",
+            isFilePath: true,
+            paramType: Diagnostics.FILE,

```

Comment:
If this is going to generalize to the LS, it seems like it might be helpful to allow directories so that each server can produce its own trace.

Reply:
I've added a bit of logic so that if you pass in a path to an existing directory, we make a timestamp-named cpuprofile in that directory; otherwise we use the input path.